### ISSUE TITLE
NIC interfaces on vmware player (NAT) can cause exported apps to hang up

### ISSUE THREAD

**Issue by rbreaves** [89luca89/distrobox#228]
**Describe the bug**
When I run Firefox from my Arch based distrobox running on podman it literally takes 2-3 minutes to start up - I have a Ryzen 7 5800x 8 core CPU (2 cores given to the VM on vmware, 4gb of ram), system has 64gb of ram in total, but I don't think that is the issue, only 2gb is in use my VM.

**To Reproduce**
Steps to reproduce the behavior

```
curl -s https://raw.githubusercontent.com/89luca89/distrobox/main/install | sudo sh
sudo apt install podman
distrobox-create --name arch --image docker.io/library/archlinux:latest
distrobox-enter arch

sudo pacman -S git neofetch 

Pre-compiled
# update
sudo pacman -Syy
sudo pacman -S appmenu-gtk-module xapp
# For proper youtube codec support
sudo pacman -S patch libunistring pkgconf make gnutls
# clone yay repo
git clone https://aur.archlinux.org/yay-git.git
 sudo pacman -S --needed base-devel
cd ./yay-git
makepkg -si
# Note many keyserver urls suck, this one works
gpg --keyserver keyserver.ubuntu.com --recv-keys 85F86E317555BECC1C2184BF2C45BA09ABC5D7DA
# For proper youtube codec support
yay -Syu ffmpeg-compat-58
yay -Syu firefox-appmenu-bin
distrobox-export --app firefox  
```

**Expected behavior**
Firefox to open nearly instantly as it does on a low powered HP Chromebox G1 that I have... a freakin celeron and 4gb of physical ram runs better than this thing.

**Logs**
Run the commands with `--verbose` and post the log here as a file upload
Attach also the output of `podman logs` or `docker logs`, possibly with `--latest` flag

Not seeing anything useful there...

**Desktop (please complete the following information):**
 - OS: Ubuntu Budgie
 - Version 21.10

**Additional context**
None that I can think of.

**Comment by rbreaves** [comment]
Found the problem.. I ran `tail -f /var/log/sys/log` and just watched it.. eventually the NIC interface goes down and comes back.. that is when the app loads. To force this I can literally toggle my internet connection and it will load right up. Why this happens.. I have no idea.

```
Apr  8 00:07:39 ubuntu podman[1988]: 2022-04-08 00:07:39.980485207 -0500 CDT m=+1.406918198 container exec eef608284043794f46d998fa8407ad899253fcf4c11d069ff7a50790a8298ee1 (image=docker.io/library/archlinux:latest, name=arch)
Apr  8 00:07:40 ubuntu systemd[1]: systemd-fsckd.service: Deactivated successfully.
Apr  8 00:07:40 ubuntu systemd-timesyncd[667]: Initial synchronization to time server 91.189.89.199:123 (ntp.ubuntu.com).
Apr  8 00:07:43 ubuntu systemd[1]: systemd-timedated.service: Deactivated successfully.
Apr  8 00:07:46 ubuntu gsd-media-keys[1351]: Failed to create proxy for screencast: Error calling StartServiceByName for org.gnome.Shell.Screencast: Timeout was reached
Apr  8 00:07:51 ubuntu systemd[1]: systemd-hostnamed.service: Deactivated successfully.
Apr  8 00:07:55 ubuntu systemd[1]: systemd-localed.service: Deactivated successfully.
Apr  8 00:09:50 ubuntu budgie-panel.desktop[2249]: _IceTransSocketUNIXConnect: Cannot connect to non-local host ubuntu
Apr  8 00:09:50 ubuntu budgie-panel.desktop[2249]: _IceTransSocketUNIXConnect: Cannot connect to non-local host ubuntu
Apr  8 00:10:19 ubuntu kintotray.desktop[1679]: /home/rreaves/.config/kinto/kintotray.py:303: DeprecationWarning: AppIndicator3.Indicator.set_icon is deprecated
Apr  8 00:10:19 ubuntu kintotray.desktop[1679]:   self.indicator.set_icon(os.environ['HOME']+'/.config/kinto/kinto.svg')
Apr  8 00:10:20 ubuntu kernel: [  195.004482] e1000: ens33 NIC Link is Down
Apr  8 00:10:20 ubuntu dbus-daemon[1187]: [session uid=1000 pid=1187] Activating via systemd: service name='org.freedesktop.portal.Desktop' unit='xdg-desktop-portal.service' requested by ':1.125' (uid=1000 pid=2267 comm="/usr/lib/firefox/firefox " label="unconfined")
Apr  8 00:10:20 ubuntu systemd[1166]: Starting Portal service...

```

**Comment by 89luca89** [comment]
> Found the problem.. I ran `tail -f /var/log/sys/log` and just watched it.. eventually the NIC interface goes down and comes back.. that is when the app loads. To force this I can literally toggle my internet connection and it will load right up. Why this happens.. I have no idea.
> 
> ```
> Apr  8 00:07:39 ubuntu podman[1988]: 2022-04-08 00:07:39.980485207 -0500 CDT m=+1.406918198 container exec eef608284043794f46d998fa8407ad899253fcf4c11d069ff7a50790a8298ee1 (image=docker.io/library/archlinux:latest, name=arch)
> Apr  8 00:07:40 ubuntu systemd[1]: systemd-fsckd.service: Deactivated successfully.
> Apr  8 00:07:40 ubuntu systemd-timesyncd[667]: Initial synchronization to time server 91.189.89.199:123 (ntp.ubuntu.com).
> Apr  8 00:07:43 ubuntu systemd[1]: systemd-timedated.service: Deactivated successfully.
> Apr  8 00:07:46 ubuntu gsd-media-keys[1351]: Failed to create proxy for screencast: Error calling StartServiceByName for org.gnome.Shell.Screencast: Timeout was reached
> Apr  8 00:07:51 ubuntu systemd[1]: systemd-hostnamed.service: Deactivated successfully.
> Apr  8 00:07:55 ubuntu systemd[1]: systemd-localed.service: Deactivated successfully.
> Apr  8 00:09:50 ubuntu budgie-panel.desktop[2249]: _IceTransSocketUNIXConnect: Cannot connect to non-local host ubuntu
> Apr  8 00:09:50 ubuntu budgie-panel.desktop[2249]: _IceTransSocketUNIXConnect: Cannot connect to non-local host ubuntu
> Apr  8 00:10:19 ubuntu kintotray.desktop[1679]: /home/rreaves/.config/kinto/kintotray.py:303: DeprecationWarning: AppIndicator3.Indicator.set_icon is deprecated
> Apr  8 00:10:19 ubuntu kintotray.desktop[1679]:   self.indicator.set_icon(os.environ['HOME']+'/.config/kinto/kinto.svg')
> Apr  8 00:10:20 ubuntu kernel: [  195.004482] e1000: ens33 NIC Link is Down
> Apr  8 00:10:20 ubuntu dbus-daemon[1187]: [session uid=1000 pid=1187] Activating via systemd: service name='org.freedesktop.portal.Desktop' unit='xdg-desktop-portal.service' requested by ':1.125' (uid=1000 pid=2267 comm="/usr/lib/firefox/firefox " label="unconfined")
> Apr  8 00:10:20 ubuntu systemd[1166]: Starting Portal service...
> ```

Hi @rbreaves this does not seem like something distrobox does, probably VM related?
How are the performance of ff installed on the VM itself?

**Comment by rbreaves** [comment]
Once it starts the performance is fine, but I think I just resolved my issue - I had to set the NIC to bridge mode (I also disabled a 2nd NIC I had as well, but really unused - for private ip stuff locally as well.. just did not need it).

To fix the bridging issue on Windows I had to go under the network adapters area of Windows then properties on the VMware NIC interface and enable the Bridge protocol option.. for whatever reason vmware player was not turning that on when I specified it. So yea.. an odd vmware configuration bug that is definitely related to the NAT interface.. I just double checked it, going back to NAT Firefox hangs again and then back to bridged it is fine.

I guess I am closing this, but there is some sorta bug with Firefox on NAT vmware interfaces that only impacts podman or containerized versions of the app - causing a severe start up delay that can last for a few or several minutes (basically the NIC interface eventually crashes and restarts even while on the NAT and the app will finish starting because the interface had gone down temporarily - note that the app will work fine even on NAT once the NAT returns).
